lib/src/chart/line_chart/line_chart.dart

```diff
@@ -47,17 +51,299 @@ class _LineChartState extends AnimatedWidgetBaseState<LineChart> {
+      // Shift the range to the right
+      final shiftedMinX = defaultMinX;
+      final shiftedMaxX = newMaxX + delta;
+      // If shiftedMaxX exceeds defaultMaxX, clamp it again (if needed)
+      // ... similar logic if the new range surpasses defaultMaxX
+      newMinX = shiftedMinX;
+      newMaxX = math.min(shiftedMaxX, defaultMaxX);
+    }
+
+    if (newMaxX > defaultMaxX) {
+      final delta = newMaxX - defaultMaxX;
+      // Shift the range to the left
+      final shiftedMaxX = defaultMaxX;
+      final shiftedMinX = newMinX - delta;
+      // If shiftedMinX falls below defaultMinX, clamp it again
+      newMaxX = shiftedMaxX;
+      newMinX = math.max(shiftedMinX, defaultMinX);
+    }
+
+    // Similarly for the Y values
+    if (newMinY < defaultMinY) {
+      final delta = defaultMinY - newMinY;
+      newMinY = defaultMinY;
+      newMaxY = math.min(newMaxY + delta, defaultMaxY);
+    }
+
+    if (newMaxY > defaultMaxY) {
+      final delta = newMaxY - defaultMaxY;
+      newMaxY = defaultMaxY;
+      newMinY = math.max(newMinY - delta, defaultMinY);
+    }
+
+    setState(() {
+      currentMinX = newMinX;
+      currentMaxX = newMaxX;
+      currentMinY = newMinY;
+      currentMaxY = newMaxY;
+    });
+  }
+
+  // Convert a pixel coordinate (localPosition) to data coordinates
+  // given the current min/max and size of the chart.
+  Offset pixelsToData(Offset pixelPos) {
+    final dxRatio = pixelPos.dx / _chartSize.width;
+    final dyRatio = pixelPos.dy / _chartSize.height;
+
+    // Assuming Y increases upwards in data, but screen Y increases downward.
+    // We'll invert Y mapping.
+    final dataX = currentMinX + dxRatio * currentDataWidth;
+    final dataY = currentMaxY - dyRatio * currentDataHeight;
+
+    return Offset(dataX, dataY);
+  }
+
+  LineChartData _transformData(LineChartData data) {
+    final isTransformed = currentMinX != data.minX ||
+        currentMaxX != data.maxX ||
+        currentMinY != data.minY ||
+        currentMaxY != data.maxY;
+
+    final newData = data.copyWith(
+      minX: currentMinX,
+      maxX: currentMaxX,
+      minY: currentMinY,
+      maxY: currentMaxY,
+      clipData: isTransformed ? const FlClipData.all() : null,
+    );
+
+    return newData;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final transformedData = _transformData(_data);
+    _lineChartDataTween = LineChartDataTween(
+      begin: transformedData,
+      end: _data,
+    );
     return AxisChartScaffoldWidget(
-      chart: LineChartLeaf(
-        data: _withTouchedIndicators(_lineChartDataTween!.evaluate(animation)),
-        targetData: _withTouchedIndicators(showingData),
-        key: widget.chartRendererKey,
+      chart: LayoutBuilder(
+        builder: (context, constraints) {
+          _chartSize = constraints.biggest;
+          return Listener(
+            onPointerSignal: _onPointerSignal,
+            child: GestureDetector(
+              onDoubleTap: resetDataAndSetState,
```

Comment:
Regarding this part, I think it makes more sense to handle the interaction in the renderer (just like we do for other interactions). Then, we can listen to the touch events and manage the double-tap callback.
This way, we keep the consistency

Reply:
Good point, will refactor accordingly 👍

Reply:
Done in 2d0c5e8